src/dom/snapshot/snapshot-view.ts

```diff
@@ -312,7 +314,14 @@ class SnapshotView extends DOMView<SnapshotViewState, SnapshotViewData> {
 	protected _navigateToSelector(selector: Selector, options: NavigateOptions = {}) {
 		let range = this.toDisplayedRange(selector);
 		if (range) {
-			getStartElement(range)?.scrollIntoView(options);
+			let elem = getStartElement(range);
+			if (!elem) return;
+			elem.scrollIntoView(options);
+			// Remember which node was navigated to for screen readers to place
+			// virtual cursor on it later. Used for navigating between sections in the outline.
+			debounceUntilScrollFinishes(this._iframeDocument, 200).then(() => {
```

Comment:
On a general scroll event, we want to clear all virtual cursor target variables, otherwise, one would navigate the outline, scroll, click on the document with a mouse, and the focus would jump to our focused element. For this reason, before almost every `setA11yVirtualCursorTarget`, we wait for `debounceUntilScrollFinishes` - to make sure that the node is set once the scrolling is done. In case of snapshot, you would navigate the outline, the document would scroll, then `await debounceUntilScrollFinishes(this._iframeDocument, 100)` would return, we'd set the target, and sometimes a random scroll event would fire after that, which would clear the node (as if the user scrolled the document themselves).

I agree that different timeouts are not very obvious so instead, I went back to using a default timeout everywhere but added a tweak to `setA11yVirtualCursorTarget` where if the target was set within the last 0.5 second, it cannot be cleared. It takes care of that edge case as well, just maybe in a bit more constructive manner.

But correct me if I overthought this and there is some better way to undo all this virtual cursor target focusing on scroll of a mouse (or click or anything like that) without such overhead.